tests/unit/test_session.py

```diff
@@ -1624,6 +1624,150 @@ def test_stop_tuning_job_client_error(sagemaker_session):
     )
 
 
+def test_train_with_sagemaker_config_injection_empty_profiler_config(sagemaker_session):
+    """
+    Tests disableProfiler property is injected into a profiler_config object when the property does not exist explicitly
+    :param sagemaker_session:
+    :return: None
+    """
+    sagemaker_session.sagemaker_config = SAGEMAKER_CONFIG_TRAINING_JOB
+
+    in_config = [
+        {
+            "ChannelName": "training",
+            "DataSource": {
+                "S3DataSource": {
+                    "S3DataDistributionType": "FullyReplicated",
+                    "S3DataType": "S3Prefix",
+                    "S3Uri": S3_INPUT_URI,
+                }
+            },
+        }
+    ]
+
+    out_config = {"S3OutputPath": S3_OUTPUT}
+
+    resource_config = {
+        "InstanceCount": INSTANCE_COUNT,
+        "InstanceType": INSTANCE_TYPE,
+        "VolumeSizeInGB": MAX_SIZE,
+    }
+
+    stop_cond = {"MaxRuntimeInSeconds": MAX_TIME}
+    RETRY_STRATEGY = {"MaximumRetryAttempts": 2}
+    hyperparameters = {"foo": "bar"}
+    TRAINING_IMAGE_CONFIG = {
+        "TrainingRepositoryAccessMode": "Vpc",
+        "TrainingRepositoryAuthConfig": {
+            "TrainingRepositoryCredentialsProviderArn": "arn:aws:lambda:us-west-2:1234567897:function:test"
+        },
+    }
+
+    profiler_config = {}
+    sagemaker_session.train(
+        image_uri=IMAGE,
+        input_mode="File",
+        input_config=in_config,
+        job_name=JOB_NAME,
+        output_config=out_config,
+        resource_config=resource_config,
+        profiler_config=profiler_config,
+        hyperparameters=hyperparameters,
+        stop_condition=stop_cond,
+        metric_definitions=METRIC_DEFINITONS,
+        use_spot_instances=True,
+        checkpoint_s3_uri="s3://mybucket/checkpoints/",
+        checkpoint_local_path="/tmp/checkpoints",
+        enable_sagemaker_metrics=True,
+        retry_strategy=RETRY_STRATEGY,
+        training_image_config=TRAINING_IMAGE_CONFIG,
+    )
+
+    _, _, actual_train_args = sagemaker_session.sagemaker_client.method_calls[0]
+
+    expected_disable_profiler_attribute = SAGEMAKER_CONFIG_TRAINING_JOB["SageMaker"]["TrainingJob"][
+        "ProfilerConfig"
+    ]["DisableProfiler"]
+    assert (
+        actual_train_args["ProfilerConfig"]["DisableProfiler"]
+        == expected_disable_profiler_attribute
+    )
+
+
+def test_train_with_sagemaker_config_injection_partial_profiler_config(sagemaker_session):
+    """
+    Tests disableProfiler property is injected into a profiler_config object when the property does not exist explicitly
+    :param sagemaker_session:
+    :return: None
+    """
```

Comment:
Curious, what happens when `DisableProfiler` is set to `True` and `profiler_config` is provided though function input?

Reply:
The behavior of `update_nested_dictionary_with_values_from_config` is - if the profiler_config is populated with the disableProfiler attribute then that will be picked over what is present in the SDK defaults. if it is partially updated with out the value the one in the config will be used, similarly None or empty

I can write a UT to validate this as well